src/nvim/api/buffer.c

```diff
@@ -142,6 +142,13 @@ Boolean nvim_buf_attach(uint64_t channel_id,
       }
       cb.on_lines = v->data.luaref;
       v->data.integer = LUA_NOREF;
+    } else if (is_lua && strequal("on_bytes", k.data)) {
```

Comment:
Would on_data be an incorrect name? We use that elsewhere.

Reply:
This entire API is subject to change, this is just the lazy "what misc1.c does internally" hack. Probably, for consumers that opt-in to byte resolution, there should be a singe event with `pos_T start, pos_T old_end, pos_T new_end` values (+ total byte count of the deleted region, though actually we should do that first, I will open a separate PR).